Dockerfile

```
@@ -1,22 +1,43 @@
-FROM tomcat:8.5-jdk8-openjdk
-MAINTAINER geosolutions<info@geo-solutions.it>
+FROM tomcat:9-jdk11-openjdk AS mother
+LABEL maintainer="Alessandro Parma<alessandro.parma@geo-solutions.it>"
+ARG MAPSTORE_WEBAPP_SRC="./.placeholder"
+ADD "${MAPSTORE_WEBAPP_SRC}" "/mapstore/"
+
+COPY ./docker/* /mapstore/docker/
+WORKDIR /mapstore
+
+FROM tomcat:9-jdk11-openjdk
 
 # Tomcat specific options
 ENV CATALINA_BASE "$CATALINA_HOME"
-ENV JAVA_OPTS="${JAVA_OPTS}  -Xms512m -Xmx512m -XX:MaxPermSize=128m"
+ENV MAPSTORE_WEBAPP_DST="${CATALINA_BASE}/webapps"
+ENV INITIAL_MEMORY="512m"
+ENV MAXIMUM_MEMORY="512m"
+ENV JAVA_OPTS="${JAVA_OPTS}  -Xms${INITIAL_MEMORY} -Xmx${MAXIMUM_MEMORY} -XX:MaxPermSize=128m"
+ENV GEOSTORE_OVR_OPT="-Dgeostore-ovr=${CATALINA_BASE}/conf/${OVR}"
+ARG OVR=""
+ARG PG_CLIENT_VERSION=""
+
+COPY --from=mother "/mapstore/mapstore.war" "${MAPSTORE_WEBAPP_DST}/mapstore.war"
+COPY --from=mother "/mapstore/docker" "${CATALINA_BASE}/docker/"
 
-# Optionally remove Tomcat manager, docs, and examples
-ARG TOMCAT_EXTRAS=false
-RUN if [ "$TOMCAT_EXTRAS" = false ]; then \
-      find "${CATALINA_BASE}/webapps/" -delete; \
+WORKDIR ${CATALINA_BASE}
+
+# adding the env
+RUN if [ "${OVR}" = "geostore-datasource-ovr.properties" ]; then \
+        cp ${CATALINA_BASE}/docker/geostore-datasource-ovr.properties ${CATALINA_BASE}/conf; \
     fi
 
```

Add wait-for-postgres script otherwise, the container won't start correctly

```suggestion

RUN cp ${CATALINA_BASE}/docker/wait-for-postgres.sh /usr/bin/wait-for-postgres && chmod +x /usr/bin/wait-for-postgres

```

---

@randomorder can we merge this change?